cli/src/main/kotlin/cli/argument/DefaultArgumentParser.kt

```diff
@@ -0,0 +1,43 @@
+package cli.argument
+
+class DefaultArgumentParser : ArgumentParser {
+    override fun parse(args: List<String>): Map<String, String> {
+        val parsedArguments = mutableMapOf<String, String>()
+        var index = 0
+
+        while (index < args.size) {
+            val currentArgument = args[index]
+
+            if (isValidFlag(currentArgument)) {
+                if (isFlagWithArgument(currentArgument)) {
+                    parsedArguments[currentArgument] = args[index + 1]
+                    index += 2
+                    continue
+                } else {
+                    parsedArguments[currentArgument] = ""
+                    index++
+                    continue
+                }
+            }
+            index++
+        }
+
+        return parsedArguments
+    }
+
```

Comment:
What would happen in the case that is 'analyze -flagWithArgument' and it does not have an argument following the flag.

Reply:
It would brake 😅